content/terraform-plugin-framework/v1.15.x/docs/plugin/framework/resources/identity.mdx

```diff
@@ -0,0 +1,194 @@
+
+1. The identity schema only supports primitive types and list types such as:
+    - `BoolAttribute`
+    - `Float32Attribute`
+    - `Float64Attribute`
+    - `Int32Attribute`
+    - `Int64Attribute`
+    - `NumberAttribute`
+    - `StringAttribute`
+    - `ListAttribute`
+        - `ElementType: types.BoolType`
+        - `ElementType: types.Float32Type`
+        - `ElementType: types.Float64Type`
+        - `ElementType: types.Int32Type`
+        - `ElementType: types.Int64Type`
+        - `ElementType: types.NumberType`
+        - `ElementType: types.StringType`
+
+<Note>
+
+    The ordering in lists is expected to be stable and semantically relevant.
+
+</Note>
+
+2. Unlike the resource state schema, the resource identity schema will not support behaviors such as Required and Computed. Two behaviors will be allowed for each attribute: `RequiredForImport` and `OptionalForImport`, which, as named, only affect behavior when importing a resource by identity.
+ - `RequiredForImport` is set to true if an import block would need this information from the practitioner to import the resource.
+
+  - `OptionalForImport` is set to true if an import block does not require this information from the practitioner to import the resource. This can be data that would normally be defaulted from the configured provider running the import.
+
+It is expected that exactly one of `RequiredForImport` or `OptionalForImport` is set to true. Regardless of which option is used, the provider can decide exactly what data is stored in the identity during import, similar to `Computed` attributes in resource state.
+
+Example of an identity schema for an aws_s3_bucket:
+```go
+var _ resource.ResourceWithIdentity = &S3BucketResource{} // new interface
+
+func (r *S3BucketResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_s3_bucket"
+}
+
+func (r *S3BucketResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// .. existing resource schema for aws_s3_bucket
+	}
+}
+
+func (r *S3BucketResource) IdentitySchema(_ context.Context, _ resource.IdentitySchemaRequest, resp *resource.IdentitySchemaResponse) {
+	resp.IdentitySchema = identityschema.Schema{
+		Attributes: map[string]identityschema.Attribute{
+			"bucket": identityschema.StringAttribute{
+				RequiredForImport: true,
+			},
+			"account_id": identityschema.StringAttribute{
+				OptionalForImport: true,
+			},
+			"partition": identityschema.StringAttribute{
+				OptionalForImport: true,
+			},
+			"region": identityschema.StringAttribute{
+				OptionalForImport: true,
+
+			},
+		},
+	}
+}
+```
+
+## Identity Schema Usage
+
```

Comment:
Can you please add a sentence here to avoid stacking headings?

Reply:
Will remove the stack headings, need to refactor this section a bit first 👍🏻